internal/config/config.go

```diff
@@ -174,62 +194,279 @@ func IsGroupsValid(groups []GroupConfig) bool {
+		if globalErrorsConfig == nil {
+			c.Rate = DefaultErrorRate
+		} else {
+			c.Rate = globalErrorsConfig.Rate
+		}
+	}
+}
+
+// Sorts in-place and removes duplicates from the specified slice.
+func sortAndRemoveDuplicates(s []string) []string {
+	slices.Sort(s)
+	return slices.Compact(s)
+}
+
 type MethodConfig struct {
 	Name      string        `yaml:"method"`
 	Threshold time.Duration `yaml:"threshold"`
 }
 
+func (c *MethodConfig) isMethodConfigValid(passiveLatencyChecking bool) bool {
+	if c == nil {
+		return true
+	}
+
+	if passiveLatencyChecking {
+		// TODO(polsar): Validate the method name: https://ethereum.org/en/developers/docs/apis/json-rpc/
+		return !strings.EqualFold(c.Name, PassiveLatencyCheckMethod)
+	}
+
+	return true
+}
+
+// LatencyConfig
+// TODO(polsar): Add the minimum number of latencies in the detection window required to apply the threshold.
+// TODO(polsar): Add other aggregation options. Currently, the average of latencies in the detection windows is used.
 type LatencyConfig struct {
+	MethodLatencyThresholds map[string]time.Duration
+
 	Methods   []MethodConfig `yaml:"methods"`
 	Threshold time.Duration  `yaml:"threshold"`
 }
 
+func (c *LatencyConfig) merge(globalConfig *LatencyConfig) {
+	if globalConfig == nil {
+		return
+	}
+
+	for method, latencyThreshold := range globalConfig.MethodLatencyThresholds {
+		if _, exists := c.MethodLatencyThresholds[method]; !exists {
+			c.MethodLatencyThresholds[method] = latencyThreshold
+		}
+	}
+}
+
+func (c *LatencyConfig) getLatencyThreshold(globalConfig *LatencyConfig) time.Duration {
+	if c.Threshold <= time.Duration(0) {
+		var threshold time.Duration
+
+		// The latency threshold is not configured or invalid, so use the global config's value or the default.
+		if globalConfig != nil {
+			threshold = globalConfig.getLatencyThreshold(nil)
+		} else {
+			threshold = DefaultMaxLatency
+		}
+
+		// The next time this method is called on the same LatencyConfig instance, this field will be set, and
+		// we simply return its value.
+		c.Threshold = threshold
+
+		return threshold
+	}
+
+	return c.Threshold
+}
+
+func (c *LatencyConfig) getLatencyThresholdForMethod(method string) time.Duration {
+	if latency, exists := c.MethodLatencyThresholds[method]; exists {
+		return latency
+	}
+
+	return DefaultMaxLatency
+}
+
+func (c *LatencyConfig) initialize(globalConfig *RoutingConfig) {
+	c.MethodLatencyThresholds = make(map[string]time.Duration)
+
+	if c.Methods == nil {
+		return
+	}
+
+	var globalLatencyConfig *LatencyConfig
+	if globalConfig != nil {
+		globalLatencyConfig = globalConfig.Latency
+	}
+
+	for _, method := range c.Methods {
+		var threshold time.Duration
+
+		if method.Threshold <= time.Duration(0) {
+			// The method's latency threshold is not configured or invalid.
+			if c.Threshold <= time.Duration(0) && globalLatencyConfig != nil {
+				// Use the top-level value.
+				threshold = globalLatencyConfig.getLatencyThresholdForMethod(method.Name)
```

Comment:
Do we want to use the default latency in this case?

e.g. 
```
# Default: 10000

global:
   - threshold: 7000
   - methods: 
       -  method: eth_getLogs
           threshold: 15000

chain
   - threshold: 5000
   - methods:
       -   method: eth_getTxReceipt
            threshold: -1
```

do we want eth_getTxReceipt to have a threshold of 5000, 7000, 10000? currently we're using 10000, but I think we want to use 5000 right?

Reply:
Correct me if i'm wrong: `c.Threshold > time.Duration(0)` will be false because `c.Threshold` hasn't been set yet. `c.Threshold` is only set after calling `getLatencyThreshold` once, right?

This is related to my comment: `Can we explicitly set c.Threshold in this initialize method before setting the method overrides?`